### ISSUE TITLE
Query return extended observable directly

### ISSUE THREAD

**Issue by anymaniax** [ngneat/query#25]
I saw that when you execute useQuery currently it returns an object with a result$ property.

Did you think about the possibility to return an observable directly that you extend with some other properties?

Like [here](https://github.com/anymaniax/azahi/blob/main/projects/azahi/src/lib/utils.ts#L233)

**Comment by NetanelBasal** [comment]
I actually start with this approach, but it doesn't work with pipeable operators. For example:

```ts
// This will break the type
const projects$ = service.getProjects().pipe(tap())
```

**Comment by NetanelBasal** [comment]
https://github.com/ngneat/query/commit/41538f77c6eef3aafebea77e52a0bfe8f55b4c64#diff-1bef14e2637a11619849bb4e431011a04387dcb25bd49abd10f39a286a0609c5L104

**Comment by anymaniax** [comment]
I remember now I had this problem too 😅.

**Comment by anymaniax** [comment]
I am gonna check a bit today If I found something for the `usePreviousData`. But overall it's pretty nice! I hope that people will like and use this library!

**Comment by NetanelBasal** [comment]
My goal was to touch the bare minimum in the source code and provide the best DX we can have.

**Comment by NetanelBasal** [comment]
I removed the need for unsubscribing explicitly when using the updateQueryKey.

**Comment by anymaniax** [comment]
Exactly what I wanted too and I think it's the best approach. Just a bit sad about the pagination because all the rest is pretty cool and straightforward

**Comment by NetanelBasal** [comment]
Yes, but it's not that bad for what you get. Moreover, not everyone needs it.
